cmd/backfillContractDiffs.go

```diff
@@ -0,0 +1,88 @@
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/makerdao/vulcanizedb/libraries/shared/factories/storage"
+	"github.com/makerdao/vulcanizedb/libraries/shared/storage/backfill"
+	"github.com/makerdao/vulcanizedb/utils"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+var (
+	backfillContractDiffsAddress     string
+	backfillContractDiffsBlockNumber int64
+	backfillContractDiffsAddressFlag = "all-contract-values-address"
+	backfillContractDiffsBlockFlag   = "all-contract-values-block-number"
+)
+
+var backfillContractDiffsCmd = &cobra.Command{
+	Use:   "backfillContractDiffs",
+	Short: "Backfills all diffs for the given contract starting at the given block.",
+	Long: fmt.Sprintf(`Backfills diffs belonging to the given contract starting at the given block.
+
+Use: ./vulcanizedb backfillContractDiffs --config=<config.toml> --%s=<block number> --%s=<address>`, backfillContractDiffsBlockFlag, backfillContractDiffsAddressFlag),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		SubCommand = cmd.CalledAs()
+		LogWithCommand = *logrus.WithField("SubCommand", SubCommand)
+		LogWithCommand.Infof("Backfilling diffs for contract %d", backfillContractDiffsBlockNumber)
+
+		if backfillContractDiffsAddress == "" {
+			return fmt.Errorf("SubCommand %v: %s argument is required and no value was given", SubCommand, backfillContractDiffsAddressFlag)
+		}
+
+		validationErr := validateBlockNumberArg(backfillContractDiffsBlockNumber, backfillContractDiffsBlockFlag)
+		if validationErr != nil {
+			return validationErr
+		}
+
+		backfillDiffsErr := backfillContractDiffs(backfillContractDiffsAddress, backfillContractDiffsBlockNumber)
+		if backfillDiffsErr != nil {
+			return fmt.Errorf("SubCommand %v: Failed to backfill diffs for contract %v. Err: %v", SubCommand, backfillContractDiffsAddress, backfillDiffsErr)
+		}
+		return nil
+	},
+}
+
+func init() {
+	backfillContractDiffsCmd.Flags().StringVarP(&backfillContractDiffsAddress, backfillContractDiffsAddressFlag, "a", "", "address whose diffs will be backfilled")
+	backfillContractDiffsCmd.Flags().Int64VarP(&backfillContractDiffsBlockNumber, backfillContractDiffsBlockFlag, "b", -1, "block number at which to start backfilling diffs for the given contract")
+	rootCmd.AddCommand(backfillContractDiffsCmd)
+}
+
+func backfillContractDiffs(address string, startingBlock int64) error {
+	blockChain := getBlockChain()
+	db := utils.LoadPostgres(databaseConfig, blockChain.Node())
+	_, storageInitializers, _, exportTransformersErr := exportTransformers()
+	if exportTransformersErr != nil {
+		return fmt.Errorf("SubCommand %v: exporting transformers failed: %v", SubCommand, exportTransformersErr)
+	}
+
+	if len(storageInitializers) == 0 {
+		return fmt.Errorf("SubCommand %v: no storage transformers found in the given config", SubCommand)
+	}
+
+	latestBlock, latestBlockErr := blockChain.LastBlock()
```

Comment:
I think this is a solid starting point and am happy to merge as is, but seeing this last block param makes me wonder if we might want to add stories for a couple possibilities - 
1) configure the last block - if you started watching the contract at block n, you know you only need to backfill from deployment to n (rather than to the head of the chain)
2) continuously check for most recent block until done - if you're backfilling before transformers are configured to watch that contract, maybe you want to keep polling the head of the chain and checking the newest blocks as it moves forward. Idk, maybe it doesn't make sense to backfill before you've got a transformer checking new blocks as part of execute 🤔

Reply:
Yeah, both those use-cases make sense to me. I'll put some stories in the backlog

Reply:
1) done

2) https://makerdao.atlassian.net/browse/VDB-1259